contrib/Joshua/scripts/localClusterStart.sh

```diff
@@ -17,373 +18,385 @@ AUDITLOG="${AUDITLOG:-/tmp/audit-cluster.log}"
-	else
-		log "Forcibly killed FDB Server process (${FDBSERVERID})"
-	fi
-	return "${status}"
+  local status=0
+
+  # Add an audit entry, if enabled
+  if [ "${AUDITCLUSTER}" -gt 0 ]; then
+    printf '%-15s (%6s)  Stopping cluster %-20s (%6s): %s\n' "$(date +'%Y-%m-%d %H:%M:%S')" "${$}" "${FDBCLUSTERTEXT}" "${FDBSERVERID}" >> "${AUDITLOG}"
+  fi
+  if [ -z "${FDBSERVERID}" ]; then
+    log 'FDB Server process is not defined'
+    let status="${status} + 1"
+  elif ! kill -0 "${FDBSERVERID}"; then
+    log "Failed to locate FDB Server process (${FDBSERVERID})"
+    let status="${status} + 1"
+  elif "${BINDIR}/fdbcli" -C "${FDBCONF}" --exec "kill; kill ${FDBCLUSTERTEXT}; sleep 3" --timeout 120 &>> "${LOGDIR}/fdbcli-kill.log"
+  then
+    # Ensure that process is dead
+    if ! kill -0 "${FDBSERVERID}" 2> /dev/null; then
+      log "Killed cluster (${FDBSERVERID}) via cli"
+    elif ! kill -9 "${FDBSERVERID}"; then
+      log "Failed to kill FDB Server process (${FDBSERVERID}) via cli or kill command"
+      let status="${status} + 1"
+    else
+      log "Forcibly killed FDB Server process (${FDBSERVERID}) since cli failed"
+    fi
+  elif ! kill -9 "${FDBSERVERID}"; then
+    log "Failed to forcibly kill FDB Server process (${FDBSERVERID})"
+    let status="${status} + 1"
+  else
+    log "Forcibly killed FDB Server process (${FDBSERVERID})"
+  fi
+  return "${status}"
 }
 
 # Start the server running.
 function startFdbServer
 {
-	local status=0
-
-	# Add an audit entry, if enabled
-	if [ "${AUDITCLUSTER}" -gt 0 ]; then
-		printf '%-15s (%6s)  Starting cluster %-20s\n' "$(date +'%Y-%m-%d %H:%M:%S')" "${$}" "${CLUSTERSTRING}" >> "${AUDITLOG}"
-	fi
-
-	if [ "${status}" -ne 0 ]; then
-		:
-	elif ! displayMessage "Starting Fdb Server"
-	then
-		log 'Failed to display user message'
-		let status="${status} + 1"
-
-	elif ! "${BINDIR}/fdbserver" --knob_disable_posix_kernel_aio=1 -C "${FDBCONF}" -p "${CLUSTERSTRING}" -L "${LOGDIR}" -d "${WORKDIR}/fdb/${$}" &> "${LOGDIR}/fdbserver.log" &
-	then
-		log "Failed to start FDB Server"
-		let status="${status} + 1"
-	else
-		FDBSERVERID="${!}"
-	fi
-
-	if [ -z "${FDBSERVERID}" ]; then
-		log "FDB Server start failed because no process"
-		let status="${status} + 1"
-	elif ! kill -0 "${FDBSERVERID}" ; then
-		log "FDB Server start failed because process terminated unexpectedly"
-		let status="${status} + 1"
-	fi
-
-	return ${status}
+  local status=0
+
+  # Add an audit entry, if enabled
+  if [ "${AUDITCLUSTER}" -gt 0 ]; then
+    printf '%-15s (%6s)  Starting cluster %-20s\n' "$(date +'%Y-%m-%d %H:%M:%S')" "${$}" "${FDBCLUSTERTEXT}" >> "${AUDITLOG}"
+  fi
+
+  if ! displayMessage "Starting Fdb Server"
+  then
+    log 'Failed to display user message'
+    let status="${status} + 1"
+
+  else
+    "${BINDIR}/fdbserver" --knob_disable_posix_kernel_aio=1 -C "${FDBCONF}" -p "${FDBCLUSTERTEXT}" -L "${LOGDIR}" -d "${WORKDIR}/fdb/${$}" &> "${LOGDIR}/fdbserver.log" &
+    if [ "${?}" -ne 0 ]
```

Comment:
This was a bit confusing to me, as I initially thought we would be checking the return code from `fdbserver`.  After talking to @AlvinMooreSr, I realized that this is to check whether the process was successfully sent to the background.  It would be nice if we had a one-line comment here.